pkg/kinflate/transformers/overlay.go

```diff
@@ -59,39 +65,143 @@ func (o *overlayTransformer) Transform(baseResourceMap resource.ResourceCollecti
 		switch {
 		case runtime.IsNotRegisteredError(err):
 			// Use JSON merge patch to handle types w/o schema
-			baseBytes, err := json.Marshal(base.Data)
+			merged, err = jsonMergePatch(base.Data.Object, overlay.Data.Object, jsonpatch.MergePatch)
 			if err != nil {
 				return err
 			}
-			patchBytes, err := json.Marshal(overlay.Data)
+		case err != nil:
+			return err
+		default:
```

Comment:
I'm really not convinced that this very weird switch is the most readable. Especially with err being defined in multiple different places :-)

Reply:
> I'm really not convinced that this very weird switch is the most readable.

Suggestions?

> Especially with err being defined in multiple different places

Not sure if I understand it.
It is switching on the same err which is returned by `versionedObj, err := scheme.Scheme.New(gvkn.GVK)`.